### PR TITLE
docs: continue reorganizing 'Advanced QL'

### PR DESCRIPTION
**Pull request by jf205**
Partially fixes https://github.com/github/product-documentation/issues/2188.

This PR continues the work started in #2583. I'm trying to reorganize the [Advanced QL section](https://help.semmle.com/QL/learn-ql/advanced/advanced-ql.html). Hopefully we'll be able to get rid of that section altogether soon.

### Changes in this PR

- Move the relevant detail from [Determining the most specific types of a variable](https://help.semmle.com/QL/learn-ql/advanced/determining-specific-types-variables.html) into the topic about debugging queries.
- Delete [Constraining types](https://help.semmle.com/QL/learn-ql/advanced/constraining-types.html) The content doesn't seem to fit anywhere. I'll probably add a redirect to the section in the handbook about [casts](https://help.semmle.com/QL/ql-handbook/expressions.html#casts).

### TODO

- Add appropriate redirects 
- Move [Monotonic aggregates in QL](https://help.semmle.com/QL/learn-ql/advanced/monotonic-aggregates.html) into the handbook, but I need some help from a QL writer to fill in some gaps in my knowledge before doing that.

### CODE REVIEW

File: docs/language/learn-ql/writing-queries/debugging-queries.rst [hunk 1]
```diff
@@ -62,6 +62,28 @@ is preferred over::
...
+   from Expr e, Callable c
+   where
+       c.getDeclaringType().hasQualifiedName("my.namespace.name", "MyClass")
+       and c.getName() = "c"
+       and e.getEnclosingCallable() = c
+   select e, e.getAQlClass()
+
+The result of this query is a list of the most specific types of every ``Expr`` in that function. You will see multiple results for expressions that are represented by more than one type, so it will likely return a very large table of results.
+
+Use ``getAQlClass()`` as a debugging tool, but don't include it in the final version of your query, as it slows down performance.
```
Comment (shati-patel) [inline docs/language/learn-ql/writing-queries/debugging-queries.rst]: Maybe this would be more visible as a note/warning?

**Comment by jf205** [comment]
@max-schaefer over to you for a technical review, if you have a moment please

**Comment by jf205** [comment]
Since these are fairly uncontroversial changes, I'd like to get this merged now as it's blocking https://github.com/github/product-documentation/issues/2157

@max-schaefer if you do have time to review, feel free to add comments here and I'll address them in a follow-up PR.

@shati-patel, before we merge I'll retarget this against rc/1.23 so that the changes become visible when we update help.semmle.com from that branch.

**Comment by jf205** [comment]
Actually, I'm going to open a fresh PR. Retargetting introduces many conflicts. Closing this now.

**Comment by jf205** [comment]
Replaced by https://github.com/Semmle/ql/pull/2876.

**Pull request closed without merging**